index.d.ts

```diff
@@ -123,7 +116,8 @@ declare module '@primer/components' {
   export const BaseStyles: React.FunctionComponent<BaseStylesProps>
 
   export interface BorderBoxProps extends CommonProps, LayoutProps, BorderProps, BoxProps {
```

Comment:
I believe all of these are actually defined in `StyledSystem.BordersProps`, `StyledSystem.BoxShadowProps` , and `StyledSystem.BorderRadiusProps` which are defined in `BorderProps`. So I think this can just be

```suggestion
  export interface BorderBoxProps extends CommonProps, LayoutProps, BorderProps, BoxProps {}
```

Reply:
We might also remove `BaseProps` from the `BorderProps` definition since it's included in `CommonProps` in the `BorderBox` type definition. but you'd need to check that any other component that's extending `BorderProps` is also extending `CommonProps`

Reply:
Sorry none if this was really added in your PR! Just figure we should clean up the types as we go 🙌

Reply:
> We might also remove BaseProps from the BorderProps definition

Turns out that `BorderBox` is the only component that inherits from `BorderProps`.

I started to make the change you suggested, but all the other interfaces based on `styled-system` props also include `BaseProps`. The more I look at it, the more I want the interfaces we define in `index.d.ts` to exactly match the `propTypes` we define in the component files. That way, it's easy to see that everything is in sync as everything matches. I'll probably do this in a different PR though.